src/Symfony/Component/Yaml/Yaml.php

```diff
@@ -41,21 +41,28 @@ class Yaml
      * @param string $input                  Path to a YAML file or a string containing YAML
      * @param bool   $exceptionOnInvalidType True if an exception must be thrown on invalid types false otherwise
      * @param bool   $objectSupport          True if object support is enabled, false otherwise
+     * @param bool   $objectForMap           True if maps should return a stdClass instead of array()
+     * @param bool   $timestampAsDateTime    True if timestamps must be parsed as DateTime objects rather than Unix timestamps (integers)
      *
      * @return array The YAML converted to a PHP array
      *
      * @throws ParseException If the YAML is not valid
      *
-     * @deprecated The ability to pass file names to the Yaml::parse method is deprecated since version 2.2 and will be removed in 3.0. Pass the YAML contents of the file instead.
+     * @deprecated The ability to pass file names to the Yaml::parse method is deprecated since version 2.7 and will be removed in 3.0. Pass the YAML contents of the file instead.
+     * @deprecated The ability to pass $timestampAsDateTime = false to the Yaml::parse method is deprecated since version 2.8. The argument will be removed in 3.0. Pass true instead.
      *
      * @api
      */
-    public static function parse($input, $exceptionOnInvalidType = false, $objectSupport = false)
+    public static function parse($input, $exceptionOnInvalidType = false, $objectSupport = false, $objectForMap = false, $timestampAsDateTime = false)
     {
+        if (!$timestampAsDateTime) {
```

Comment:
Well, if you cannot avoid triggering the warning for people using this method in the most common way (``Yaml::parse($yaml)``), you should not trigger it at all.
Currently, you will force everyone to pass all arguments explicitly to change the default to avoid the deprecation warning, which is a no-go